From 352bab2669196063987362bdea4095d4e1cc367e Mon Sep 17 00:00:00 2001
From: Codrin Ciubotariu <codrin.ciubotariu@microchip.com>
Date: Wed, 23 Mar 2022 19:59:09 +0200
Subject: [PATCH 042/471] ASoC: dt-bindings: Document Microchip's ASRC

Add DT bindings for the new Microchip ASRC embedded in sama7g5 SoCs.

Signed-off-by: Codrin Ciubotariu <codrin.ciubotariu@microchip.com>
---
 .../bindings/sound/microchip,asrc.yaml        | 117 ++++++++++++++++++
 1 file changed, 117 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/microchip,asrc.yaml

diff --git a/Documentation/devicetree/bindings/sound/microchip,asrc.yaml b/Documentation/devicetree/bindings/sound/microchip,asrc.yaml
new file mode 100644
index 000000000000..6e93f9c64d8c
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/microchip,asrc.yaml
@@ -0,0 +1,117 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/microchip,asrc.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Asynchronous Sample Rate Converter
+
+maintainers:
+  - Codrin Ciubotariu <codrin.ciubotariu@microchip.com>
+
+description:
+  The Asynchronous Sample Rate Converter (ASRC) converts the sample rate of an
+  incoming audio frame without affecting quality. It supports input and output
+  sampling rates up to 192 kHz. It is made of 4 independent digital signal
+  processing modules (DSP)
+
+properties:
+  compatible:
+    const: microchip,sama7g5-asrc
+
+  reg:
+    maxItems: 1
+
+  "#sound-dai-cells":
+    const: 1
+
+  interrupts:
+    maxItems: 1
+
+  clocks:
+    items:
+      - description: Peripheral Bus Clock
+      - description: Generic Clock
+
+  clock-names:
+    items:
+      - const: pclk
+      - const: gclk
+
+  dmas:
+    description: RX and TX DMA Channels
+    minItems: 8
+    maxItems: 8
+
+  dma-names:
+    items:
+      - const: rx0
+      - const: tx0
+      - const: rx1
+      - const: tx1
+      - const: rx2
+      - const: tx2
+      - const: rx3
+      - const: tx3
+
+  microchip,triggers:
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    minItems: 1
+    maxItems: 16
+    description: |
+      List of phandles pointing to other audio peripherals that can generate
+      a trigger to ASRC. The trigger is used by the ASRC to compute the needed
+      sampling rate.
+
+  microchip,trigger-indexes:
+    $ref: /schemas/types.yaml#/definitions/uint32-array
+    minItems: 1
+    maxItems: 16
+    description: |
+      An array of external (to ASRC) trigger indexes. Position of an entry
+      determines to which trigger phandle (from the 'trigger' property) the
+      value reffers to. e.g. the first value in 'trigger-indexes' will
+      correspond to the first phande in 'triggers', the second value from
+      'trigger-indexes' will correspond to the second phandle in 'triggers',
+      etc.
+
+required:
+  - compatible
+  - reg
+  - "#sound-dai-cells"
+  - interrupts
+  - clocks
+  - clock-names
+  - dmas
+  - dma-names
+  - microchip,triggers
+  - microchip,trigger-indexes
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/at91.h>
+    #include <dt-bindings/dma/at91.h>
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+
+    asrc: sound@e1610000 {
+        compatible = "microchip,sama7g5-asrc";
+        reg = <0xe1610000 0x1000>;
+        interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>;
+        #sound-dai-cells = <1>;
+        dmas = <&dma0 AT91_XDMAC_DT_PERID(55)>,
+               <&dma0 AT91_XDMAC_DT_PERID(56)>,
+               <&dma0 AT91_XDMAC_DT_PERID(57)>,
+               <&dma0 AT91_XDMAC_DT_PERID(58)>,
+               <&dma0 AT91_XDMAC_DT_PERID(59)>,
+               <&dma0 AT91_XDMAC_DT_PERID(60)>,
+               <&dma0 AT91_XDMAC_DT_PERID(61)>,
+               <&dma0 AT91_XDMAC_DT_PERID(62)>;
+        dma-names = "rx0", "tx0", "rx1", "tx1", "rx2", "tx2", "rx3", "tx3";
+        clocks = <&pmc PMC_TYPE_PERIPHERAL 30>, <&pmc PMC_TYPE_GCK 30>;
+        clock-names = "pclk", "gclk";
+        microchip,triggers = <&i2s0>, <&i2s1>, <&pdmc0>, <&pdmc1>, <&ssc0>,
+                             <&ssc1>, <&spdiftx>, <&spdifrx>;
+        microchip,trigger-indexes = <1>, <2>, <3>, <4>, <5>, <8>, <9>, <10>;
+    };
-- 
2.34.1

